Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonCancelDialogButton.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Drawing;
+using System.Windows.Forms;
+
+namespace Krypton.Toolkit.Suite.Extended.Buttons
+{
+    [ToolboxBitmap(typeof(KryptonButton))]
+    public class KryptonCancelDialogButton : KryptonButton
+    {
+        private KryptonForm _parent;
+
+        public KryptonForm ParentWindow { get => _parent; set { _parent = value; Invalidate(); } }
+
+        public KryptonCancelDialogButton()
+        {
+            DialogResult = DialogResult.Cancel;
+
+            ParentChanged += KryptonCancelDialogButton_ParentChanged;
```

Comment:
I have found it is best practice to put the equivalent release `-=` of these "events" ina dispose: i.e.
`ParentChanged -= KryptonCancelDialogButton_ParentChanged;`